tests/queries/0_stateless/02559_sorted_full_join.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT -mn -q "
+DROP TABLE IF EXISTS t1;
+DROP TABLE IF EXISTS t2;
+DROP TABLE IF EXISTS t3;
+DROP TABLE IF EXISTS t4;
+DROP TABLE IF EXISTS ta1;
+DROP TABLE IF EXISTS ta2;
+
+CREATE TABLE t1 (x UInt64, y UInt64, val UInt64) ENGINE = MergeTree ORDER BY (x, y) PARTITION BY (x % 2)
+AS SELECT sipHash64(number, '11') % 100, sipHash64(number, '12') % 100, sipHash64(number, '13') % 100 FROM numbers(1000);
+
+CREATE TABLE t2 (a UInt64, b UInt64) ENGINE = MergeTree ORDER BY (a, b) PARTITION BY (a % 2)
+AS SELECT sipHash64(number, '21') % 100, sipHash64(number, '22') % 100 FROM numbers(1000);
+
+CREATE TABLE t3 (x UInt64, y UInt64) ENGINE = MergeTree ORDER BY (x, y) PARTITION BY (x % 2)
+AS SELECT sipHash64(number, '21') % 100, sipHash64(number, '22') % 100 FROM numbers(1000);
+
+CREATE TABLE t4 (y UInt64, x UInt64) ENGINE = MergeTree ORDER BY (y, x) PARTITION BY (y % 2)
+AS SELECT sipHash64(number, '21') % 100, sipHash64(number, '22') % 100 FROM numbers(1000);
+
+CREATE TABLE ta1 (a1 UInt64, a2 UInt64, a3 UInt64, a4 UInt64, a5 UInt64, a6 UInt64, a7 UInt64, a8 UInt64, a9 UInt64, a10 UInt64)
+ENGINE = MergeTree ORDER BY (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) PARTITION BY (a1 % 2)
+AS SELECT
+    sipHash64(number, 'a11') % 2, sipHash64(number, 'a12') % 2, sipHash64(number, 'a13') % 2, sipHash64(number, 'a14') % 2, sipHash64(number, 'a15') % 2,
+    sipHash64(number, 'a16') % 2, sipHash64(number, 'a17') % 2, sipHash64(number, 'a18') % 2, sipHash64(number, 'a19') % 2, sipHash64(number, 'a20') % 2
+FROM numbers(1000);
+
+CREATE TABLE ta2 (a1 UInt64, a2 UInt64, a3 UInt64, a4 UInt64, a5 UInt64, a6 UInt64, a7 UInt64, a8 UInt64, a9 UInt64, a10 UInt64)
+ENGINE = MergeTree ORDER BY (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) PARTITION BY (a1 % 2)
+AS SELECT
+    sipHash64(number, 'a21') % 2, sipHash64(number, 'a22') % 2, sipHash64(number, 'a23') % 2, sipHash64(number, 'a24') % 2, sipHash64(number, 'a25') % 2,
+    sipHash64(number, 'a26') % 2, sipHash64(number, 'a27') % 2, sipHash64(number, 'a28') % 2, sipHash64(number, 'a29') % 2, sipHash64(number, 'a30') % 2
+FROM numbers(1000);
+"
+
+CLIENT_ARGS='--join_algorithm=full_sorting_merge --optimize_sorting_by_input_stream_properties=1 --optimize_read_in_order=1 --max_threads=8 --max_block_size=128'
+
+# test_query <expected_number_of_sort_steps> <query>
+function test_query {
+    expected_result=${1}
+    query_str="${2}"
+
+    # count the number of MergeSortingTransform's in the pipeline
+    result=$( $CLICKHOUSE_CLIENT $CLIENT_ARGS -q "EXPLAIN PIPELINE ${query_str}" )
+    sort_count=$( echo "$result" | grep 'MergeSortingTransform' | wc -l )
```

Comment:
There is no check that reading in order was applied i.e. , for example, check for `MergeTreeInOrder` in case of pipeline. In general, absense of `MergeSortingTransform` doesn't imply that reading was in order of sorting key. Please do explicit check

Reply:
Changed to expected number of reads in order